lib/utils/validateTypes.js

```diff
@@ -2,7 +2,7 @@
 
 /**
  * Checks if the value is a boolean or a Boolean object.
- * @param {any} value
+ * @param {unknown} value
  * @returns {boolean}
```

Comment:
Just in idea, don't know if it does make sense and helpful. Make all these functions into type guards:

```
 * @param {unknown} value
 * @returns {value is boolean}
```

Here's how they work as type guards: [Playground Link](https://www.typescriptlang.org/play?filetype=js#code/PQKhCgAIUhhALApgYwNYGdIEsBmkAuSkAbgIYA2ArotpqZAHaUC2ARogE6QD2X9Aci3ZdurAFYp8AOigwAAgAdSHUs0gBvSg1QNuAdwYBfEhWqzIcjonyUODTOrJUaWTEzadDs4OBxbk+FjcDLSCHhwAFE7UAJQa4ACQVjZ2BACeCojceNE0ALwFkADk7sJFkAA+FSbO2Pb4pAzIWXhhwgDc4F7goPL4GTTqjWnGID7Iwej4kAAmpA2QeZAAjAAsneAT9ZDISGiLoUKcEXMNMRu4kBGubcenpDFx6olb6NzkiFLk3ADmJ-MPTrdV7vT7fP73c5AA). Check `data` type in both `console.log`. Inside `if` `data` becomes known type. These functions could catch more issues, theoretically.

Here how it currently is: [Playground Link](https://www.typescriptlang.org/play?filetype=js#code/PQKhCgAIUhhALApgYwNYGdIEsBmkAuSkAbgIYA2ArotpqZAHaUC2ARogE6QD2X9Aci3ZdurAFYp8AOigwAAgAdSHUs0gBvSg1QNuAdwYBfEhWqzIcjonyUODTOtbdu5RKSOzg4HFuT4s3Ay0gmycABRkVIgAlBrgAJBWNnYEAJ4KiNx4kdSQALwFkADkTKEcRZAAPpUmUdj2+O7ImXghwgDc4Ibg4KDy+Ok06u6pxiBeyIHo+JAAJqSN+ZAAjAAsneCTDZDISGhLWOht4fON0Ru4kGGHxxxhp6TRseoJW+guiFLk3ADm9wuPTrdTZTD5fX7-M7tIA). `data` inside `if` is still `any`.

Reply:
@hudochenkov Nice idea! I don't know we can use the feature with JSDoc! Changed 24a6f295. 👍🏼